source/docs/software/advanced-control/state-space/state-space-flywheel-walkthrough.rst

```diff
@@ -1,4 +1,123 @@
+
+.. todo:: link to allwpilib example
+
+Why use state-space control?
+============================
+
+Because IIC is a meme award. Thank you for coming to my ted talk
+
+.. todo:: real explication.
+
+Modeling our flywheel
+=====================
+
+The idea of modeling the :term:`dynamics` of physical systems using a system of equations is central to state-space control and modern control theory. Recall that continuous state-space systems are modeled using the following system of equations:
+
+.. math::
+    \dot{\mathbf{x}} &= \mathbf{A}\mathbf{x} + \mathbf{B}\mathbf{u} \\
+    \mathbf{y} &= \mathbf{C}\mathbf{x} + \mathbf{D}\mathbf{u} 
+
+Where :term:`x-dot` is the rate of change of the :term:`system`'s :term:`state`, :math:`\mathbf{x}` is the system's current state, :math:`\mathbf{u}` is the :term:`input` to the system, and :math:`\mathbf{y}` is the sytem's :term:`output`.
+
+Let's use this system of equations to model our flywheel in two different ways. We'll first model it using :term:`system identification` using the frc-characterization toolsuite, and then model it based on the motor and flywheel's angular moment of inertia.
+
+The first step of building up our state-space system is picking our system's states. We can pick anything we want as a state -- we could pick completely unrelated states if we wanted -- but it helps to pick states that are important. We can include :term:`hidden state`\s in our state (such as elevator velocity if we were only able to measure its position) and let our Kalman Filter estimate their value if desired. Remember that the states we choose will be driven towards their :term:`reference` by our :ref:`Linear-Quadratic Regulator <docs/software/advanced-control/state-space/state-space-intro:The Linear-Quadratic Regulator>`.
+
+For our flywheel, we care only about one state: its velocity. While we could chose to also model its acceleration, the inclusion of this state isn't necessary for our system. 
+
+Next, we identify the :term:`input`\s to our system. Inputs can be thought of as things we can put "into" our system to change its state. In the case of the flywheel (and many other single-jointed mechanisms in FRC), we have just one input: voltage applied to the motor. By choosing voltage as our input we can compensate for battery voltage sag as battery load increases. 
+
+Now that we know our flywheel's states and inputs, we need to identify the :math:`\mathbf{A}` and :math:`\mathbf{B}` matrices. The last thing we still don't know is what :math:`\mathbf{\dot{x}}`, or :term:`x-dot`, is. X-dot is the rate of change of the state, so because our state is velocity :math:`\mathbf{\dot{x}}` will be the flywheel's angular acceleration.  Because they are States x States and States x Inputs respectively, they are both matrices with 1 row and 1 column. 
```

Comment:
Consider changing the sentence to "X-dot is the rate of change of the state. Because our state is velocity..."